Objects/longobject.c

```diff
@@ -3579,7 +3563,7 @@ long_hash(PyLongObject *v)
     int sign;
 
     if (_PyLong_IsCompact(v)) {
-        x = _PyLong_CompactValue(v);
+        x = (Py_uhash_t)_PyLong_CompactValue(v);
```

Comment:
Since `Py_hash_t` != `int`:
```suggestion
        x = (Py_uhash_t)(Py_hash_t)_PyLong_CompactValue(v);
```